### PR TITLE
GLSP-975 Ensure generic GLSP Server implementation

### PR DESCRIPTION
**Pull request by tortmayr**
- Introduce a common subclass for jsonrpc based ServerLaunchers
- Move out jsonrpc implementation details from `GLSPServer` implementation into the `JsonRpcGLSPServerLauncher`. This ensures that the Server implementation itself is independent from the chosen communication channel and can reused for other use cases like direct invocation in a node backend or orthogonal communication channels like REST or gRPC.
- Update to latest @eclipse-glsp/protocol version and consume shared server interface types from there

- Ensure that the `@eclipse-glsp-examples/workflow-server` package is properly indexed
- Add entry points for browser and node 
- Move rpc-server initialization out of index files into app files. 

Also: Update example applications to ensure that promise rejections/errors are properly catched.

Part of https://github.com/eclipse-glsp/glsp/issues/975

Requires https://github.com/eclipse-glsp/glsp-client/pull/245

Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: packages/server/src/common/utils/console-logger.ts
Comment (sgraband) [inline packages/server/src/common/utils/console-logger.ts:1]: I guess this should replace `packages/server/src/browser/di/console-logger`?
The old file however was not removed as far as i can see. Could you do that?

File: packages/server/src/common/session/client-session-listener.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
+
```
Comment (sgraband) [inline packages/server/src/common/session/client-session-listener.ts:16]: Minor, but this just introduces an empty line, which is not required by the linter afaik.

File: examples/workflow-server/browser.d.ts
Comment (sgraband) [inline examples/workflow-server/browser.d.ts:1]: Should this file be checked in?
If yes, could we maybe add a comment for why this is needed?

File: examples/workflow-server/browser.js
Comment (sgraband) [inline examples/workflow-server/browser.js:1]: Should this file be checked in?
If yes, could we maybe add a comment for why this is needed?

File: examples/workflow-server/node.d.ts
Comment (sgraband) [inline examples/workflow-server/node.d.ts:1]: Should this file be checked in?
If yes, could we maybe add a comment for why this is needed?

File: examples/workflow-server/node.js
Comment (sgraband) [inline examples/workflow-server/node.js:1]: Should this file be checked in?
If yes, could we maybe add a comment for why this is needed?

**Review by sgraband — COMMENTED** [review]
Thank you for the changes! They look good and work great! I just have a few questions/comments inline below. Could you take a look at them?

File: packages/server/src/common/utils/console-logger.ts
Comment (tortmayr) [reply packages/server/src/common/utils/console-logger.ts:1]: Good catch. Yes exactly, the intention was to move the generic `ConsoleLogger` from the browser into the common package. I will remove the redundant file.

File: examples/workflow-server/browser.d.ts
Comment (sgraband) [reply examples/workflow-server/browser.d.ts:1]: Makes sense. Thank you for the explanation. I was simply confused by the location of the files, but i don't think we need a comment then.

**Pull request merged**